core/types/src/attributes.rs

```diff
@@ -79,31 +79,44 @@ impl Display for Attributes {
 
 impl Display for AttributeFlags {
     fn fmt(&self, f: &mut Formatter) -> core::fmt::Result {
-        for (idx, (name, flag)) in self.iter_names().enumerate() {
-            if *self & flag == flag {
-                if idx >= 1 {
-                    write!(f, " | ")?;
-                }
-                write!(f, "{}", name)?;
-            }
-        }
-        Ok(())
+        format_bitflags(self, f)
     }
 }
 
 impl Display for ExtendedFeatureRequestMask {
     fn fmt(&self, f: &mut Formatter) -> core::fmt::Result {
-        for (idx, (name, flag)) in self.iter_names().enumerate() {
-            if *self & flag == flag {
-                if idx >= 1 {
-                    write!(f, " | ")?;
-                }
-                write!(f, "{}", name)?;
-            }
-        }
+        format_bitflags(self, f)
+    }
+}
 
-        Ok(())
+/// Formats the provided bitflags as text
+///
+/// Any bits taht aren't part of a contained flag will be formatted as a hex number.
+///
+/// Example output with two known flags and some bits that didn't pertain to a tag
+///
+///    FLAG1 | FLAG2 | 0x0000_0000_FF00_0000
+///
+fn format_bitflags<F: Flags>(bitflags: &F, f: &mut Formatter) -> core::fmt::Result
```

Comment:
moved to a function so both `AttributeFlags` and `ExtendedFeatureRequestMask` would behave the same.